File: PRACTICAS/PRACTICA_ACTUAL/RUBY/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'casilla.rb'
require_relative 'jugador.rb'

module Civitas
  class Casilla_juez < Casilla
    
    def initialize(nombre,num_casilla_carcel)
      super(nombre)
      @carcel = num_casilla_carcel
    end
    
     def recibe_jugador(iactual,todos)
  if (jugador_correcto(iactual,todos))
       informe(iactual,todos)
       Diario.instance.ocurre_evento("El jugador ha caido en la casilla juez, va a ser trasladado a la carcel")
        todos[iactual].encarcelar(@carcel)
      end
    end
    
     #Override
    def to_s
      puts "#{super.to_s} Es de tipo Juez y la carcel se encuentra en la posicion #{@carcel}"
    end
    
    
  end
end

File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author javierramirezp
 */
public class CasillaJuez extends CasillaDescanso{
    private static int carcel;
    
    CasillaJuez(String nombre, int carcel){
    
        super(nombre);
        this.carcel=carcel;
    
    }
    @Override
     void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual, todos)){
            informe(iactual, todos);
            todos.get(iactual).encarcelar(carcel);
        }              
    }
    @Override
    public String toString(){   
        return " llamada" + nombre + " con la carcel en la posicion "+carcel;
    }
}
